compiler/plugins/input/StableHLO/stablehlo-iree/PluginRegistration.cpp

```diff
@@ -116,6 +123,7 @@ struct StableHLOSession
   void populateCustomInputConversionTypes(StringSet<> &typeMnemonics) override {
     typeMnemonics.insert("stablehlo");
     typeMnemonics.insert("stablehlo_xla");
+    typeMnemonics.insert("vhlo");
```

Comment:
I really wish there weren't now *three* ways to invest things that are nominally xla-ish. I feel like it should be possible to collapse all of these down to one input pipeline that does the right thing. It might even work like that right now with no further changes (ie. I would expect that VHLO does no damage is there are no VHLO ops).

Reply:
+1

Reply:
hi @stellaraccident, One solution I can think of to remove this typeMnemoic insertion is to enable add the VHLO conversion as default (put the call `::mlir::stablehlo::createStablehloDeserializePipeline(passManager)` here: https://github.com/bviyer/iree/blob/main/compiler/plugins/input/StableHLO/Conversion/Passes.cpp#L43) and remove the `else if (typeMnemonics == "vhlo")`. 

From what I understand (and I can double-check), if the input-code is not VHLO the pass just becomes a NOp. Please advice if this sounds OK with you.

Reply:
I'm not sure I can advise: I've not looked at the stablehlo stuff in a long time. But I expect that the different entry points are doing work that is all safe to do all of the time, so there should just be one stablehlo input type that doors all of it. Those extra passes are kind of taking care of different "vintages" and quirks but no user is going to understand that.

Reply:
AFAIK, one can just have one here. The existing two one could merge and just always run the detupling one (it's inefficient but fine - auto pipeline tried to populate some info that could allow skipping parts not needed and same could be done), similarly one can ideally just run vhlo deserialization (but, as Stella said, that requires it behaves as one would expect on non-VHLO ops). In that case we just unconditionally run it internal to the StableHLO pipeline.

Reply:
@jpienaar Is it OK if I make that change in a different PR? I would like to keep this just VHLO conversion related (if it is OK with you)